controllers/workspace_controller_sshkey.go

```diff
@@ -10,78 +10,71 @@ import (
 	tfc "github.com/hashicorp/go-tfe"
 )
 
-func (r *WorkspaceReconciler) getSSHKeyIDByName(ctx context.Context, w *workspaceInstance) (string, error) {
-	SSHKeyName := w.instance.Spec.SSHKey.Name
-
+func (w *workspaceInstance) getSSHKeyIDByName(ctx context.Context) (string, error) {
 	listOpts := &tfc.SSHKeyListOptions{
 		ListOptions: tfc.ListOptions{
-			PageSize: maxPageSize,
+			PageNumber: 1,
+			PageSize:   maxPageSize,
 		},
 	}
 	for {
-		SSHKeys, err := w.tfClient.Client.SSHKeys.List(ctx, w.instance.Spec.Organization, listOpts)
+		sshKeyList, err := w.tfClient.Client.SSHKeys.List(ctx, w.instance.Spec.Organization, listOpts)
 		if err != nil {
 			return "", err
 		}
-		for _, s := range SSHKeys.Items {
-			if s.Name == SSHKeyName {
+
+		for _, s := range sshKeyList.Items {
+			if s.Name == w.instance.Spec.SSHKey.Name {
 				return s.ID, nil
 			}
 		}
-		if SSHKeys.NextPage == 0 {
+
+		if sshKeyList.NextPage == 0 {
 			break
 		}
-		listOpts.PageNumber = SSHKeys.NextPage
+		listOpts.PageNumber = sshKeyList.NextPage
 	}
 
-	return "", fmt.Errorf("ssh key ID was not found for ssh key name %q", SSHKeyName)
+	return "", fmt.Errorf("ssh key ID was not found for SSH key name %q", w.instance.Spec.SSHKey.Name)
 }
 
-func (r *WorkspaceReconciler) getSSHKeyID(ctx context.Context, w *workspaceInstance) (string, error) {
-	specSSHKey := w.instance.Spec.SSHKey
-
-	if specSSHKey.Name != "" {
-		w.log.Info("Reconcile SSH Key", "msg", "getting ssh key ID by name")
-		return r.getSSHKeyIDByName(ctx, w)
+func (w *workspaceInstance) getSSHKeyID(ctx context.Context) (string, error) {
+	if w.instance.Spec.SSHKey.Name != "" {
```

Comment:
I think this needs a nil check on `SSHKey`. It seems to be a pointer.

Reply:
Refactored it, thanks!